base/applications/rapps/gui.cpp

```diff
@@ -536,7 +536,8 @@ CMainWindow::OnCommand(WPARAM wParam, LPARAM lParam)
                 break;
 
             case ID_REFRESH:
-                UpdateApplicationsList(SelectedEnumType);
+                bReload = IsInstalledEnum(SelectedEnumType); // Force installed refresh from the registry
```

Comment:
If what this `IsInstalledEnum` does is enumerating & checking all apps, I think this is unneccessary because `UpdateApplicationsList` will properly reload the installed apps just fine when forcing `bReload = TRUE`.

Reply:
> If what `IsInstalledEnum` does is enumerating & checking all apps

No, it's super fast, just `return (x >= ENUM_INSTALLED_MIN && x <= ENUM_INSTALLED_MAX);`

Reply:
From the code, I assume this range check is for preventing refresh operation on "Available for Installation" page.
However, I think that allowing refresh operation on "Available for Installation" page is not pointless because it allows the user to manually modify the database and test the result without having to restart RAPPS.

Reply:
I'm fine with it either way

Reply:
OK. The refresh fix part can be achieved right now by my PR [#6864](https://github.com/reactos/reactos/pull/6864). I'll merge that PR soon. After that, reverting only the changes in `CMainWindow::OnCommand` function should fix all the conflicts with this PR.